x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/exchange/ExchangeResponse.java

```diff
@@ -10,15 +10,26 @@
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
 import org.elasticsearch.compute.data.Page;
+import org.elasticsearch.core.AbstractRefCounted;
 import org.elasticsearch.core.Nullable;
+import org.elasticsearch.core.RefCounted;
+import org.elasticsearch.core.Releasable;
 import org.elasticsearch.transport.TransportResponse;
 
 import java.io.IOException;
 import java.util.Objects;
 
-public final class ExchangeResponse extends TransportResponse {
+public final class ExchangeResponse extends TransportResponse implements Releasable {
+    private final RefCounted counted = AbstractRefCounted.of(this::close);
```

Comment:
Adding Releasable and RefCounted here is kinda magic, but I like it! :-)

Reply:
Yeah. I've copied it from other friends.